helper/cert/source_gen_test.go

```diff
@@ -237,3 +240,132 @@ func testGetHostname(t *testing.T) string {
 	}
 	return host
 }
+
+func TestGensource_appendCert(t *testing.T) {
+	// Construct caBundle's (old and new) to use in the test cases
+	s1 := testGenSource()
+	newBundle, err := s1.Certificate(context.Background(), nil)
+	require.NoError(t, err)
+
+	s2 := testGenSource()
+	oldBundle1, err := s2.Certificate(context.Background(), nil)
+	require.NoError(t, err)
+	s3 := testGenSource()
+	oldBundle2, err := s3.Certificate(context.Background(), nil)
+	require.NoError(t, err)
+
+	tests := map[string]struct {
+		oldCAs   []byte
+		expected []byte
+	}{
+		"no old CAs": {
+			oldCAs:   nil,
+			expected: newBundle.CACert,
+		},
+		"one old CA": {
+			oldCAs:   oldBundle1.CACert,
+			expected: append(oldBundle1.CACert, newBundle.CACert...),
+		},
+		"two old CAs": {
+			oldCAs:   append(oldBundle1.CACert, oldBundle2.CACert...),
+			expected: append(oldBundle2.CACert, newBundle.CACert...),
+		},
+		"invalid old CAs": {
+			oldCAs:   []byte("not a cert"),
+			expected: newBundle.CACert,
+		},
+	}
+
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			result, err := appendCert(tc.oldCAs, newBundle.CACert)
```

Comment:
Is it worth adding a test case where you go through an additional generation of `appendCert`? I was thinking yes because of the fact it consumes its own output (eventually), and so it would be nice to ensure by induction that it can continue forever, but after thinking about it a bit more I'm not sure if it really adds much to the coverage. Up to you if you think it has value 🙂